### DISCUSSION TITLE
Feature Request: Add a cheatsheet to show all keybindings

### DISCUSSION THREAD

**Discussion by Akianonymus** [NTBBloodbath/doom-nvim#52] [Ideas]
Maybe a doom help page which will show all the keybindings, including ctrl+* mappings.

**Comment by NTBBloodbath** [comment]
Hey, I think we definitely should add it. In fact that's something that I was thinking about while developing the 3.0 release but I didn't started it because there were some big changes in the keybindings between the 2.3 and the 3.0 releases. A lot of new keybinds were added and some others were modified.

But now that the 3.0 was released I think that we should start doing it as soon as possible, because that is something essential for moving around. Maybe a wiki entry for the keybindings inside a `Getting started` section will be really good. Let me know if you have some ideas for this.

Regards :)

**Comment by Akianonymus** [comment]
> Hey, I think we definitely should add it. In fact that's something that I was thinking about while developing the 3.0 release but I didn't started it because there were some big changes in the keybindings between the 2.3 and the 3.0 releases. A lot of new keybinds were added and some others were modified.
> 
> But now that the 3.0 was released I think that we should start doing it as soon as possible, because that is something essential for moving around. Maybe a wiki entry for the keybindings inside a `Getting started` section will be really good. Let me know if you have some ideas for this.
> 
> Regards :)

About some ideas, the cheatsheet i am talking about would show in neovim too.

Initially, this idea came from awesome wm

Here is a screenshot

![122490009-95fd9980-cffe-11eb-9676-78019aa2cd65](https://user-images.githubusercontent.com/32216346/125499270-e2d71cca-8876-4f21-b9c5-e750585b1ed0.png)

Obviously, wiki entry is nice too, but would be super handy to have it just in terminal, no need for any hassle.

And bind it to a leader + some key and make it show in which key options too ( basically which key on steroids ).

A local cheatsheet will also show any changes if applied by users. About this, instead of using predefined stuff, there should be a lua script which will parse all the keybindings we wanna show.

About parsing mappings data, :map  / :imap / :vmap without any arguments shows all the commands.

Anyway, i am pretty new to vim/nvim in general, so trying to make it friendly and exploring it's awesomeness, so don't mind if there is something dumb in above things suggested, xD.

**Comment by NTBBloodbath** [comment]
Ah got it. It's "not possible" by default in Neovim, but I thought about spawning a floating window to display the keybinds with a keystroke. Gustavo (one of our maintainers :) ) told me that the `orgmode.nvim` plugin has something similar, I'll take a look at it later so we can recreate something like that.

In fact that's a really neat idea, I'm pretty sure that it's gonna be useful to have it inside Neovim. Thank you for the idea!

**Comment by Akianonymus** [comment]
> Ah got it. It's "not possible" by default in Neovim, but I thought about spawning a floating window to display the keybinds with a keystroke. Gustavo (one of our maintainers :) ) told me that the `orgmode.nvim` plugin has something similar, I'll take a look at it later so we can recreate something like that.
> 
> In fact that's a really neat idea, I'm pretty sure that it's gonna be useful to have it inside Neovim. Thank you for the idea!

A floating window should be good enough.

I was initially thinking of a telescope style floating window, with a search bar, so we could easily search through it. Although i am not sure telescope has a grid like layout or even usable for this.

Anyway, glad i could help.

**Comment by NTBBloodbath** [comment]
Hmm [Pocco](https://github.com/Pocco81) just told me about a plugin called [cheatsheet.nvim](https://github.com/sudormrfbin/cheatsheet.nvim) that achieves exactly what you've said btw, and it's a better idea imo. Replicating it will be a neat journey (in the case that we can't use it directly, otherwise we will use it to avoid wasting time) :)

**Comment by Akianonymus** [comment]
> Hmm [Pocco](https://github.com/Pocco81) just told me about a plugin called [cheatsheet.nvim](https://github.com/sudormrfbin/cheatsheet.nvim) that achieves exactly what you've said btw, and it's a better idea imo. Replicating it will be a neat journey (in the case that we can't use it directly, otherwise we will use it to avoid wasting time) :)

That seems to be static, it is trying to parse some predefined text file.( or i am missing something )

**Comment by NTBBloodbath** [comment]
I think you're right. To fit our needs we should make our own, but I'll like to take that `cheatsheet.nvim` as inspiration for the start point (I've never did telescope integrations before, I'll need to take a look at).

Maybe we should move this issue to a discussion btw? I think it will make more sense :thinking:

**Comment by NTBBloodbath** [comment]
Hey, it's been a long time.

I think that I got something with [this new plugin](https://github.com/lazytanuki/nvim-mapper), still some things to be polished but here we go. Hope I can push this to development branch today :)

![image](https://user-images.githubusercontent.com/36456999/127124574-a88af2c4-f4cd-41b9-8225-53a688fdfdbf.png)

**Reply by Akianonymus** [reply]
Looks nice

**Reply by NTBBloodbath** [reply]
Pog, it's already pushed to development branch :)
